dashboard/app/models/script.rb

```diff
@@ -245,13 +245,13 @@ class << self
 
     def all_scripts
       Rails.cache.fetch('valid_scripts/all') do
-        Script.all
+        Script.all.to_a.freeze
```

Comment:
Have you confirmed that this works? I only ask because I remember trying something like this and finding some weird behavior where the freeze didn't seem to persist in the cache. Here's something I tried in staging rails console just now:

```
[staging] dashboard > original = [1, 2].freeze
=> [1, 2]
[staging] dashboard > original[0] = 3
Traceback (most recent call last):
        8: from ./bin/rails:9:in `<main>'
        7: from ./bin/rails:9:in `require'
        6: from /var/lib/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands.rb:18:in `<top (required)>'
        5: from /var/lib/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/commands_tasks.rb:49:in `run_command!'
        4: from /var/lib/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/commands_tasks.rb:78:in `console'
        3: from /var/lib/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/console_helper.rb:9:in `start'
        2: from /var/lib/gems/2.5.0/gems/railties-5.0.7.2/lib/rails/commands/console.rb:65:in `start'
        1: from (irb):2
FrozenError (can't modify frozen Array)

[staging] dashboard > Rails.cache.fetch('foo') { original }
=> [1, 2]

[staging] dashboard > Rails.cache.fetch('foo')[0] = 3
=> 3

[staging] dashboard > Rails.cache.fetch('foo')
=> [3, 2]
```

Not sure if I'm interpreting it right and no idea why it might work like this.

Reply:
Oh weird.  I thought I verified that Script.all.freeze worked as expected but after looking into this more, I guess it didn't.

I looked into the behavior above and strangely, Rails does a deep copy ([ref1](https://github.com/rails/rails/blob/v5.0.7.2/activesupport/lib/active_support/cache/memory_store.rb#L128), [ref2](https://github.com/rails/rails/blob/v5.0.7.2/activesupport/lib/active_support/cache.rb#L666)) of the object before putting it into the cache but it doesn't protect that copy from being further modified.

It looks like freezing the object fetched from the cache is effective so I'm going to change the code to do that.  Thanks for catching this!